packages/date-picker/src/DatePicker/DatePickerComponent/DatePickerComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  forwardRef,
+  KeyboardEventHandler,
+  TransitionEventHandler,
+  useEffect,
+  useRef,
+} from 'react';
+import { ExitHandler } from 'react-transition-group/Transition';
+
+import { useBackdropClick, useForwardedRef } from '@leafygreen-ui/hooks';
+import { keyMap } from '@leafygreen-ui/lib';
+
+import { useDatePickerContext } from '../../shared/components/DatePickerContext';
+import { DatePickerInput } from '../DatePickerInput';
+import { DatePickerMenu } from '../DatePickerMenu';
+import { useSingleDateContext } from '../SingleDateContext';
+
+import { DatePickerComponentProps } from './DatePickerComponent.types';
+
+export const DatePickerComponent = forwardRef<
+  HTMLDivElement,
+  DatePickerComponentProps
+>(({ ...rest }: DatePickerComponentProps, fwdRef) => {
+  const { isOpen, menuId, disabled, isSelectOpen } = useDatePickerContext();
+  const {
+    refs,
+    value,
+    closeMenu,
+    menuTriggerEvent,
+    handleValidation,
+    getHighlightedCell,
+  } = useSingleDateContext();
+
+  const formFieldRef = useForwardedRef(fwdRef, null);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  useBackdropClick(closeMenu, [formFieldRef, menuRef], isOpen);
```

Comment:
should not fire when a select is open

Reply:
I believe we `stopPropagation` within the menu element. Are you seeing a bug?

Reply:
Yes. if you click outside an open date picker when a select is open both close

Reply:
@bruugey Can you open a ticket for this if you haven't already ?

Reply:
https://jira.mongodb.org/browse/LG-3836